File: day6/ex12.zig
const std = @import("std");

const stdout = std.io.getStdOut().writer();

const data = @embedFile("test_data.txt");

const DAYS = 18;

pub fn main() !void {
    var fish_count = [_]usize{0} ** 9;

    var data_it = std.mem.tokenize(u8, data, ",\n");
    while (data_it.next()) |fish_str| {
        const fish = try std.fmt.parseUnsigned(u4, fish_str, 10);
        fish_count[fish] += 1;
    }

    var i: usize = 0;
    while (i < DAYS) : (i += 1) {
        std.mem.rotate(usize, &fish_count, 1);
        fish_count[6] += fish_count[8];
    }

    var answer: usize = 0;
    for (fish_count) |count| answer += count;
    try stdout.print("answer: {}\n", .{answer});
}

File: day6/ex12.py
with open("test_data.txt", "r") as fd:
    data = list(map(int, fd.read().strip().split(",")))
    fish_count = [data.count(d) for d in range(9)]

days = 18
for _ in range(days):
    fish_count = fish_count[1:] + fish_count[:1]
    fish_count[6] += fish_count[8]

print("answer:", sum(fish_count))
